library/Admin/Gutenberg/Blocks/BlockManager.php

```diff
@@ -139,12 +139,34 @@ public function renderContainerCallback($block)
         if ($this->validateFields($block['data']) || in_array($block['name'], $this->noValidationRequired)) {
             $data['style'] = [];
 
-            if (!empty($data['color'])) {
+            if (!empty($data['color']) && $data['background_color_type'] != 'gradient') {
                 $data['style'][] = "background-color:{$data['color']}";
             }
             if (!empty($data['text_color'])) {
                 $data['style'][] = "color:{$data['text_color']}";
             }
+
+            if (!empty($data['background_gradient']) && $data['background_color_type'] == 'gradient')  {
```

Comment:
An improvement on readability would be make the conditional into it's separate function.
Do not commit this suggestion, the `blockHasBackgroundGradient` function does not exist.
```suggestion
            if ($this->blockHasBackgroundGradient($data))  {
```